sdk/eventhub/azure-eventhub/samples/async_samples/recv_with_custom_starting_position_async.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+An example to show receiving events from Event Hub partitions with custom starting position asynchronously.
+"""
+import os
+import asyncio
+from azure.eventhub.aio import EventHubConsumerClient, EventHubProducerClient
+from azure.eventhub import EventData
+
+CONNECTION_STR = os.environ["EVENT_HUB_CONN_STR"]
+EVENTHUB_NAME = os.environ['EVENT_HUB_NAME']
+
+
+async def on_partition_initialize(partition_context):
+    # put your code here
+    print("Partition: {} has been initialized".format(partition_context.partition_id))
+
+
+async def on_partition_close(partition_context, reason):
+    # put your code here
+    print("Partition: {} has been closed, reason for closing: {}".format(partition_context.partition_id,
+                                                                         reason))
+
+
+async def on_error(partition_context, error):
+    # put your code here
+    print("Partition: {} met an exception during receiving: {}".format(partition_context.partition_id,
+                                                                       error))
```

Comment:
In the `on_error` callback, `partition_context` can now potentially be `None` - so we should make sure our samples demonstrate good practice and check `partition_context` before accessing `partition_context.partition_id`.

Reply:
updated :)

Reply:
updated :)